core/include/list.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+ /**
+ * @addtogroup  core_util
+ * @{
+ *
+ * @file
+ * @brief       Intrusive linked list
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef __LIST_H
+#define __LIST_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief List node structure
+ *
+ * Used as is as reference to a list.
+ *
+ * Actual list objects should have a @c list_node_t as member and then use
+ * the container_of() macro in list operations.
+ * See @ref thread_add_to_list() as example.
+ */
+typedef struct list_node {
+    struct list_node *next;     /**< pointer to next list entry */
+} list_node_t;
+
+/**
+ * @brief insert object into list
+ *
+ * @param[in] node      list node before new entry
+ * @param[in] new_node  list node to insert
+ */
+static inline void list_add(list_node_t *node, list_node_t *new_node) {
+    new_node->next = node->next;
+    node->next = new_node;
+}
+
+/**
+ * @brief removes and returns list head
```

Comment:
> (This version returns the new head, not the removed element).

Might make sense to be a little clearer in the doc (-> removes the first element in the list and returns the new head)?